### ISSUE TITLE
Clear/deleting does NOT clear form value

### ISSUE THREAD

**Issue by jjulian91** [monim67/django-bootstrap-datepicker-plus#102]
**Describe the bug**

**To Reproduce**
1. Go to any page with picker information that already exists
2. Click on clear on widget or delete the value
3. submit form
4. Data NOT cleared

**Expected behavior**
Clear button or deleting value from form to delete the value in the javascript.

**Setup Information (please complete the following information):**
 - OS: Ubuntu 22.04
 - Browser chrome
 - Browser version 115.0.5790.110
 - Python version 3.10
 - Django version 4.2
 - Bootstrap version 4
 - jQuery version 3.6.1

[x] I have followed the [configuration instructions][configuration_page] and checked out the
[common error troubleshooting][troubleshooting_page] page.

[configuration_page]: https://monim67.github.io/django-bootstrap-datepicker-plus/configure/
[troubleshooting_page]: https://django-bootstrap-datepicker-plus.readthedocs.io/en/latest/Troubleshooting.html

**Comment by christianwgd** [comment]
I've tried to reproduce the problem in my minimal example (https://github.com/christianwgd/date_time_picker_sample). Everything is working as expected. Maybe you can compare the relevant code to find out what the difference is or extend the example to show the problem in detail.

**Comment by jjulian91** [comment]
![image](https://github.com/monim67/django-bootstrap-datepicker-plus/assets/45022382/c44827c1-26b1-4f83-8183-ad53505ca69a)

![image](https://github.com/monim67/django-bootstrap-datepicker-plus/assets/45022382/255b2914-8fe6-4637-820c-a906196e5819)

**Comment by christianwgd** [comment]
Not sure, what the images should tell us. Could you please add some comments? Thanks

**Comment by christianwgd** [comment]
Yes, I think we all had understood, what your problem is. But we want to find out why. So, did you compare your code to that one in the sample? Can we see your code (view, form and template)? It's really hard to help you without any clue...

**Comment by jjulian91** [comment]
Form:
![image](https://github.com/monim67/django-bootstrap-datepicker-plus/assets/45022382/9fedf3d8-b67c-4e3e-b10e-a1d946b2dba9)

View:
![image](https://github.com/monim67/django-bootstrap-datepicker-plus/assets/45022382/7248cff2-33b6-4010-bd9a-1807c9a636ec)

Inherited View:
![image](https://github.com/monim67/django-bootstrap-datepicker-plus/assets/45022382/8b384931-83c5-43a2-9e83-247852edb416)

template:
![image](https://github.com/monim67/django-bootstrap-datepicker-plus/assets/45022382/c5e68c2c-b139-48c1-acf1-c0e4abcf6c6f)

**Comment by christianwgd** [comment]
First of all: It would be much easier to read, if you would paste the code as text and not as images, thanks.
If I want to test if your code is working, I would have to change my minimal example to work like your app. I need some time for that.
Did you ever solve #98? If so, then it would be nice if you close that. If not, are your sure you're not fail from that?

A hint on your model form: As far as I know, the _required_ property is only for simple forms. For model forms this is dependent on your model fields. These are optional in your form, if they have the "blank=True" property set. I'm not sure if you can override that with the "required" option.

**Comment by jjulian91** [comment]
> First of all: It would be much easier to read, if you would paste the code as text and not as images, thanks. If I want to test if your code is working, I would have to change my minimal example to work like your app. I need some time for that. Did you ever solve #98? If so, then it would be nice if you close that. If not, are your sure you're not failing from that?
> 
> A hint on your model form: As far as I know, the _required_ property is only for simple forms (forms.Form). For model forms this is dependent on your model fields. These are optional in your form, if they have the "blank=True" property set. I'm not sure if you can override that with the "required" option.

No #98 is not solved -- no they are not related; the issue with #98 is the CSS doesn't account for viewport size of embedded templates.
 
It should be pretty clear that the form I sent is a model form that is directly tied to the model view/update view.  The use of inline formsets here are pretty irrelevant and should be able to replicate with a simple view/form -- I only grabbed the exact example I dealt with when the page came up -- the issue seems to be that on page render of an initial value being added to the widget, the only time the value is touched is when it is "changed" to a new value, not changed to a null value.

model forms are able to be modified in way I have done it, as the super().__init__() loads the model definition, and the subsequent override of the required attr is then changed as if the model definition included blank=True -- this does not work for non nullable fields, but does work if the field is nullable as this field is.

**Comment by jjulian91** [comment]
`{% extends 'base.html' %} {% block container %}
<div class="text-black">
  {% load bootstrap4 static tags %} 
  {{ form.media }}
  <div class="d-flex align-items-center mb-3 pb-1">
    <i class="fas fa-cubes fa-2x me-3" style="color: #00bfff"></i>
    <span class="h1 fw-bold mb-0">{{ view.getTitle }} </span>
  </div>

  
  <form method="POST" enctype="multipart/form-data">
    {% csrf_token %} {% bootstrap_form form %}
    <h4 style="text-align: center">Technician Job Details</h4>
    <hr />
    {% include 'invoices/technicians/parameters.html' %}
    <h4 style="text-align: center">Technician Purchase Order Mileage</h4>
    <hr />
    {% include 'invoices/technicians/mileage.html' %}
    {% startAssessment as startNote %}
    {% if startNote %}
    <h4 style="text-align: center">Technician Start Note</h4>
    <hr />
    {% include 'invoices/technicians/notes.html' %}
    {% endif %}
    {% changeNote as changeNote %}
    {% if changeNote %}
    <h4 style="text-align: center">Technician Change Notes</h4>
    <hr />
    {% include 'invoices/technicians/changenotes.html' %}
    {% endif %}
    {% endAssessment as endNote %}
    {% if endNote %}
    <h4 style="text-align: center">Technician Finish Notes</h4>
    <hr />
    {% include 'invoices/technicians/finishnotes.html' %}
    {% endif %}
    {% if start %}
    <h4 style="text-align: center">Technician Job Start Questions</h4>
    <hr />
    {% include 'invoices/technicians/start.html' %}
    {% endif %}
    {% if finish %}
    <h4 style="text-align: center">Technician Job Finish Questions</h4>
    <hr />
    {% include 'invoices/technicians/finish.html' %}
    {% endif %}
    {% include 'audit.html' %}
    <div class="pt-1 mb-4">
      <button class="btn btn-dark btn-lg btn-block" type="submit">Save</button>
    </div>
    
  </form>
</div>
{% endblock %} {% block javascripts %}

        <script src="{% static 'generics/js/formatting-functions.js' %}"></script>


{% endblock %}
`

**Comment by monim67** [comment]
I can't reproduce the problem on widget's demo website, @jjulian91 I'd suggest check the demo website to figure out what are you doing different if you still have the issue.
